app/helpers/application_helper.rb

```diff
@@ -9,7 +9,10 @@ def master?
   end
 
   def homepage?
-    current_page?(root_path) || current_page?(versioned_root_path(version: ruby_version))
```

Comment:
Unfortunately not. When memoizing a variable, it’s important to consider if the value is expected to be falsey. If it is expected to be falsey, then you can’t use `||=`, because you’d recompute your falsey values.

Basically, a `||=` would only memoize this method if this page was indeed the home page (a truthy value). If it’s not the home page (99.9999% of the site), then it would do `false ||= false` every time you called the method.

Reply:
That being said, we could change the variable name if you’d like. I thought `@homepage` seemed like a variable that might possibly be in use or later be in use to hold some homepage object, so I went with `@is_homepage`.

Reply:
> Unfortunately not. When memoizing a variable, it’s important to consider if the value is expected to be falsey. If it is expected to be falsey, then you can’t use `||=`, because you’d recompute your falsey values.
> 
> Basically, a `||=` would only memoize this method if this page was indeed the home page (a truthy value). If it’s not the home page (99.9999% of the site), then it would do `false ||= false` every time you called the method.

Yes, and we can use `defined? @homepage`: https://karolgalanciak.com/blog/2017/05/28/ruby-memoization-%7C%7C-equals-vs-defined-syntax/

It'd be larger, like `return @foo if defined? @foo; @foo = ...`

Reply:
> That being said, we could change the variable name if you’d like. I thought `@homepage` seemed like a variable that might possibly be in use or later be in use to hold some homepage object, so I went with `@is_homepage`.

I think, when it will be in use — then we can resolve this. 

In RuboCop we have [`Naming/MemoizedInstanceVariableName` cop](https://www.rubydoc.info/gems/rubocop/RuboCop/Cop/Naming/MemoizedInstanceVariableName).

Reply:
> > That being said, we could change the variable name if you’d like. I thought `@homepage` seemed like a variable that might possibly be in use or later be in use to hold some homepage object, so I went with `@is_homepage`.
> 
> 
> 
> I think, when it will be in use — then we can resolve this. 
> 
> 
> 
> In RuboCop we have [`Naming/MemoizedInstanceVariableName` cop](https://www.rubydoc.info/gems/rubocop/RuboCop/Cop/Naming/MemoizedInstanceVariableName).

@AlexWayfer I know rubocop has that rule, but it doesn’t make sense in the context of a `?` method, because you could have 2 methods competing for the same instance variable (`homepage` and `homepage?`).

Reply:
> > Unfortunately not. When memoizing a variable, it’s important to consider if the value is expected to be falsey. If it is expected to be falsey, then you can’t use `||=`, because you’d recompute your falsey values.
> 
> > 
> 
> > Basically, a `||=` would only memoize this method if this page was indeed the home page (a truthy value). If it’s not the home page (99.9999% of the site), then it would do `false ||= false` every time you called the method.
> 
> 
> 
> Yes, and we can use `defined? @homepage`: https://karolgalanciak.com/blog/2017/05/28/ruby-memoization-%7C%7C-equals-vs-defined-syntax/
> 
> 
> 
> It'd be larger, like `return @foo if defined? @foo; @foo = ...`

I’m not sure what you’re suggesting compared to what I currently have. 

I’m open to change, I just don’t know how to twist it simpler. Problems make to solve:

- Memoize falsey values too
- Don’t accidentally memoize only half of the calculation

Reply:
> @AlexWayfer I know rubocop has that rule, but it doesn’t make sense in the context of a `?` method, because you could have 2 methods competing for the same instance variable (`homepage` and `homepage?`).

Yes, that true. But that's how it was intended: https://github.com/rubocop-hq/rubocop/issues/5648

To be honest, I don't know what `@homepage` can store.

Reply:
> I’m not sure what you’re suggesting compared to what I currently have.

Sorry, my bad. Nothing. I looked at the old code.

That's why I don't like discussions left to not the end of diff!

Reply:
> Yes, that true. But that's how it was intended: [rubocop-hq/rubocop#5648](https://github.com/rubocop-hq/rubocop/issues/5648)
And stuff like this is why I don't always agree with Rubocop.

> To be honest, I don't know what `@homepage` can store.
Me either, but since this is the `ApplicationHelper`, I don't want to assume I know what instance variables someone would define in any other helper method across all other controllers. You could make a whole new controller later on and add a `homepage` helper method to it and set `@homepage` inside that method, and now you have a collision.